File: backend/models/surge.py
from pydantic import BaseModel, Field
from typing import List, Dict, Any, Optional

class Platform(BaseModel):
    id: str
    name: str
    max_capacity: int
    typical_load_peak: int
    typical_load_offpeak: int

class Station(BaseModel):
    id: str
    name: str
    code: str
    platforms: List[Platform]

class CrowdRiskAssessment(BaseModel):
    station_id: str
    platform_id: str
    score: float
    level: str  # Normal, Elevated, Critical
    time_to_critical: Optional[int] = None
    contributing_factors: Dict[str, Any]
    calculated_at: str

File: frontend/src/types/surge.ts
export interface Platform {
  id: string;
  name: string;
  max_capacity: number;
  typical_load_peak: number;
  typical_load_offpeak: number;
}

export interface Station {
  id: string;
  name: string;
  code: string;
  platforms: Platform[];
}

export interface CrowdRiskAssessment {
  station_id: string;
  platform_id: string;
  score: number;
  level: 'Normal' | 'Elevated' | 'Critical';
  time_to_critical: number | null;
  contributing_factors: {
    platform_capacity: number;
    typical_load: number;
    expected_passengers_from_delayed_trains: number;
    delayed_trains_count: number;
    delayed_train_numbers?: string[];
    formula: string;
  };
  calculated_at: string;
}
